[realqijun] iP

realqijun opened the pull request:
# BabyGronk
> Rizzing up livy dunne 🇬🇾

If you have tons of tasks and want to a way to keep track of them, ask BabyGronk for help

BabyGronk supplies you with the ability to
* Add/Delete tasks
* Mark/Unmark tasks
* ~~Communicate with him as a chatbot~~

There are 3 types of tasks
1. Todo
2. Deadline
3. Event
---
[image](https://tr.rbxcdn.com/dfb4be8368f04e1916a79a44c60a692b/420/420/Hat/Webp)

The main method of the program
```java
  public static void  main(String[] args) {
      new BabyGronk("./data/BabyGronk.txt").run();
  }
```

According to CS2103T website
- [x] Week1
- [x] Week2
- [x] Week3
- [ ] Week4

Also visit: [Pull request](https://github.com/nus-cs2103-AY2425S1/ip/pull/182)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
change name and add greeting + goodbye messages

Commit:
add read input functionality

Commit:
add store string methods

Commit:
add tasks with mark and unmark feature

Commit:
add todo, deadline, event tasks

Commit:
modify test scripts

Commit:
add EmptyInputException

Commit:
add delete function

Commit:
disallow mark of marked tasks

Commit:
Add file reading method and logic to convert data to tasks

Data stored in .txt file cannot be converted to tasks immediately

Create a function that reads from constant string DATA <- name of file
Creates the file if it doesn't exist, and reads tasks of same format
we see when list command is used in program

Have not create function to write changes into the file

Commit:
Add save data method

Tasks were not saved after program ended previously,
saveData saved current list of tasks into the file in const DATA
overwriting the data in the file.

saveData is called when user types bye and nowhere else

Commit:
Merge branch 'branch-Level-7'

Commit:
Add CustomDate class to store date and modify Tasks to handle it

Dates were stored as String and users can input strings that are
not dates. CustomDate specifies the date format to be dd/MM/yyyy
HHmm or dd-MM-yyyy HHmm otherwise it sets it to the current Datetime

Dates are stored in the data file in the format dd/MM/yyyy HHmm by
using the toString method and they are displayed with format using
the display method with format dd of LL yyyy at HH:mm am/pm

Commit:
Merge branch 'branch-Level-8'

Commit:
Add more classes to adhere to OOP principles.

Most of the program logic was stored in BabyGronk class, which
did not adhere much to OOP style of coding.

Parser is introduced to parse user input and produce Instruction
that has the correct command as well as args formatted, otherwise,
an InvalidInputException will be thrown.

TaskList is created to store tasks and handle add and delete operations.

Ui class handles the printing of strings to stdout as well as
exiting the program.

Commit:
Merge branch 'A-MoreOOP'

Commit:
Package all class into BabyGronk.

Classes were not packaged.

Packaged everything into BabyGronk.

Please remember to organize them into proper packages when you
learn more about packaging.

Commit:
Merge branch 'A-Package'

Commit:
Move files into packages.

Files were all under 1 package.

Create several new packages to organise files.

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add JUnit tests.

Public methods can be tested for potential bugs using JUnit tests.

Commit:
Modify logic in TaskList.

TaskList was displaying 0 indexed tasks, but delete command is 1 indexed
so delete wasn't working well with tasks.

Change TaskList to display tasks 1 indexed.

Commit:
Add JavaDoc.

Program wasn't documented properly, hard to navigate through program.

Add simple javadoc to all classes to help explain the code better.

Commit:
Format code.

Code violated some format standards.

Format some code and made sure they still pass the unit tests.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add find function.

If user input too many tasks, search function may be useful to find
a task that contains a certain keyword easily.

Add find method in tasklist that uses string.contains() method to match
input string with the full task name.

Commit:
Merge branch 'branch-Level-9'

Commit:
commit before reset

Commit:
Complete javafx tutorial up to 3

testing1234567891011121314 commented on data/BabyGronk.txt:
```diff
@@ -1,7 +1,7 @@
+[E][ ] mooncake event  (from: 02/09/2024 1830 to: 02/09/2024 2030)
 [T][ ] go home
-[D][ ] apply sep  (by: 30/08/2024 1327)
+[D][ ] apply sep  (by: 25/09/2024 0000)
```
all the best!!

testing1234567891011121314 commented on src/main/java/task/TaskList.java:
```diff
@@ -67,6 +67,26 @@ public String   markTask(String[] input, boolean status) {
         return (builder.toString());
     }
 
+    public String   find(String needle) {
```
Hi there is an extra space here

Commit:
Add checkstyle xml files

Coding standards were hard to maintain and check for especially
since there are so many classes in the program.

Added checkstyle.xml and suppressions.xml that works with build.gradle
to run checkstyle on all the files conveniently.

Also fixed some style errors for some files.

testing1234567891011121314 commented on src/main/java/babygronk/CustomDate.java:
```diff
@@ -17,17 +17,19 @@ public CustomDate(String date) {
+        DateTimeFormatter formatter;
         if (args.length == 2) {
             if (args[0].contains("/")) {
                 formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
I like how you handled multiple inputs for date and time!

Commit:
Merge branch 'branch-A-CheckStyle'

testing1234567891011121314 commented on src/main/java/babygronk/CustomDate.java:
```diff
@@ -17,17 +17,19 @@ public CustomDate(String date) {
         }
     }
 
     private void parseDate(String date) throws DateTimeParseException {
```
I like that your method names are very clear!

testing1234567891011121314 reviewed (commented):
LGTM, very nice!

tohhengxing commented:
Small nitpicks

1. Inconsistent commit messages - Some are capitalised, some are not
2. Code style issues that can be resolved via a linter, not worth reviewing.

Other than that, looks fine.

tohhengxing commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+        }
+
+        if (matches.isEmpty()) {
+            return ("No matches to " + needle + " found\n");
```
Messages can actually be turned into Constants

tohhengxing commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,91 @@
+        if (input.startsWith("list")) {
+            return (new Instruction(args[0], args));
+        }
+        if (args[0].equals("mark") || args[0].equals("unmark") || args[0].equals("m") || args[0].equals("um")) {
```
It might be more legible to do

`isArgMark`, `isArgUnmark` and pass in args[0] as a param

Or better yet, define parser class for the args

tohhengxing commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,91 @@
+        }
+        if (args[0].equals("delete")) {
+            if (args.length < 2) {
+                throw new InvalidInputException("How can you not delete properly? (delete [index])\n");
```
This can be a constant

tohhengxing commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,91 @@
+                System.arraycopy(deadlineArgs, 0, deadlineRet, 1, deadlineArgs.length);
+                return (new Instruction(deadlineTemp[0], deadlineRet));
+            case "event":
+                String[] eventTemp = input.split(" ", 2);
```
Can consider refactoring this into a method

Commit:
Add fxml files for DialogBox and MainWindow

Commit:
Add final touches to GUI

GUI didn't have fxml and selected css implemented
yet, therefore it was lackluster.

Written fxml code that sets mainWindow as
fx:root and set it to a controller(whatever
these meant). Add .css files that help style
the GUI elements.

Finally, format the files with some protected
keywords(WIP), and refactor some code to qualify
for checkstyle.

Commit:
Merge branch 'branch-Level-10'

Commit:
Add assertions

Code did not have assertions to protect program
against bugs.

Add simple assertions to BabyGronk::handleInput
and CustomDate::toString.

Commit:
Refactor code

Code was a little hard to read and logic was
messy.

Broke down some functions into parts and created
an enum InvalidCommands for
InvalidInputException.

Importantly, created functions such as
isMarkCommand, isDeleteCommand, isTaskCommand
in Instruction class so that it can be used by
Parser and BabyGronk class statically. Might
need to change this later.

Commit:
Modify saveData method in Storage

Previously, storage saves the list of tasks line
by line. Now, storage saves the list of tasks
using stream's foreach method line by line.

realqijun committed:
Merge pull request #1 from realqijun/branch-A-Assertions

Add assertions

realqijun committed:
Merge branch 'master' into branch-A-CodeQuality

realqijun committed:
Merge pull request #2435246354654534 from realqijun/branch-A-CodeQuality

Refactor code

realqijun committed:
Merge branch 'master' into branch-A-Streams

realqijun committed:
Merge pull request #3 from realqijun/branch-A-Streams

Modify saveData method in Storage

Commit:
Fix saveData method in Storage class

Tasks were not being saved into the datafile.

Added invocation to bufferedwriter flush and
close methods.

This fixes the issue but more problems might
arise later.

Commit:
Add more streams

Problem: Not enough streams.

Added more streams in TaskList find and toString
methods.

Commit:
Set up CI for project

Project doesn't update whenever PR updates/
master branch updates/any other project event.

Add .yml file in .github/workflows to automate
this process.

Commit:
Merge branch 'branch-A-Streams'

Commit:
Merge branch 'branch-A-CI'

Commit:
Add more search functions

Find function used String.contains method
to match keywords.

Added an interative match expression function
that also handles for the wildcard '*' so that
users can find tasks that matches the expression.

Commit:
Merge branch 'branch-C-BetterSearch'

Commit:
Refactor code

CI feature was not working for Windows because
there was unreadable characters such as the
emojis.

Variable names in match expression function
is also not valid.

Fixed by deleting emojis and such and changing
the names of variables.

Commit:
Attempt to fix code

CI on github keeps producing compilation errors
due to errors in checkstyle or others.

Fixed some checkstyle erros and modify code
in StorageTest.

Commit:
Add test cases to TaskListTest

TaskList test was not testing TaskList class
extensively.

Added more tests to TaskListTest, and made sure
that everything was up to checkstyle standard.

Commit:
Attempt to fix StorageTest for CI

CI compile fails for Windows. Error was in
StorageTest.

Use chatGPT to modify cleanup method in
StorageTest.

Commit:
Refactor StorageTest

StorageTest is still causing problems on
Github CI.

Add AfterEach JUnit method thing to delete
testfiles after running the tests.

Commit:
Remove StorageTests for CI

Unable to find fix for CI for Windows on Github.

Remove Tests that were causing trouble.

Commit:
Add reply colors and background image to GUI

GUI previously only had simple colors and no
background.

Add colors to responses and a background image.

Commit:
Merge branch 'branch-A-BetterGui'

Commit:
Add content to hi message

Hi message was only a single word.

Add more words to hi message.

Commit:
Merge branch 'branch-A-Personality'

Commit:
Add functionality to find

Find function that uses wildcard didn't work
too well.

Add a function that splits the string and
matches expression for each element.

Commit:
Fix task deletion

Task deletion had minor bugs in the code.

Used streams to delete tasks.

Commit:
Merge branch 'branch-A-MoreErrorHandling'

Commit:
Refactor code

Parser wasn't checking for lowercased input for
'hi' and 'bye' inputs. And TaskList was returning
an empty string if the list was empty.

Modify parser to check input correctly. And add
return string for when the list is empty in
TaskList.

Commit:
Add product demo image

Program did not have a demo image to show its
functionalities.

Add Ui.png to docs folder.

Commit:
Merge branch 'branch-A-MoreErrorHandling'

Commit:
Refactor code

TaskList wasn't passing the JUnit tests due to
change in code in previous commit.

Change test to fit output of TaskList. And remove
unused import of Collectors.

Commit:
Make changes to README.md

Github pages was set up to display README on
project page.

Added some text to README.

Commit:
Make incremental changes to README

README was not completed yet.

Add more documentation of program to the README.
Also added functionality to search tasks by type.

Commit:
Merge branch 'branch-A-UserGuide'

Commit:
Add JUnit tests got Parser class

Parser class has many methods and had no tests
for them.

Replace TodoTest with ParserTest because TodoTest
didn't have much to test for. Also implemented
links in README.

Commit:
Merge branch 'branch-A-UserGuide'

Commit:
Modify README

README wasn't able to display properly on GitHub
Pages.

Changed the format of some items.

Commit:
Finalize code

Code lacked some documentation and Storage had a
minor bug.

Refactor some code and add comments to some lines
that are complex. Also changed Storage's
fileCreated method to check for directory first
then check for file existence. Then, modify
storage test to test for Storage class more
rigorously.

Commit:
Remove StorageTest class

StorageTest class was creating problems for CI
on GitHub especially for Windows OS.

Remove StorageTest class entirely.

Commit:
Rewrite JavaDoc comments
